### PR TITLE
FIle organizer script 

### PR DESCRIPTION
**Pull request by guruswarupa**
# Pull Request

## Title
FIle organizer script for basic organizing on a directory

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/utils/file-organizer.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            organize_by_date "$TARGET_DIR"
+            ;;
+        *)
+            echo "Invalid choice. Please enter 1 or 2."
+            exit 1
+            ;;
+    esac
+}
+
+# Main script execution
```
Comment (jeevithakannan2) [inline tabs/utils/file-organizer.sh:76]: checkEnv should called at all scripts as per @nnyyxxxx

Comment (nnyyxxxx) [reply tabs/utils/file-organizer.sh:76]: @jeevithakannan2 Only if any of the functions in common-script is being called.

Comment (guruswarupa) [reply tabs/utils/file-organizer.sh:76]: 😂

Comment (jeevithakannan2) [reply tabs/utils/file-organizer.sh:76]: Well we should be maintaining uniformity, As like you said. I still don't get why checkEnv should be be called with all those other functions while I need to check only for escalation, the escalation function does not contain variables or any sort of things related to other functions. That's why I am so confused

Comment (nnyyxxxx) [reply tabs/utils/file-organizer.sh:76]: > Well we should be maintaining uniformity, As like you said. I still don't get why checkEnv should be be called with all those other functions while I need to check only for escalation, the escalation function does not contain variables or any sort of things related to other functions. That's why I am so confused

I have explained it to you many times

> Only if any of the functions in common-script are being called. Then you should call checkEnv.

Comment (jeevithakannan2) [reply tabs/utils/file-organizer.sh:76]: > > Well we should be maintaining uniformity, As like you said. I still don't get why checkEnv should be be called with all those other functions while I need to check only for escalation, the escalation function does not contain variables or any sort of things related to other functions. That's why I am so confused
> 
> I have explained it to you many times
> 
> > Only if any of the functions in common-script are being called. Then you should call checkEnv.

What I am asking is the escalation function does not depend on checEnv and other function, then what is purpose of calling other functions? You didn't explain why we are calling other functions all you said was it should be called and don't debate

Comment (guruswarupa) [reply tabs/utils/file-organizer.sh:76]: continue the debate in discord please.

Comment (jeevithakannan2) [reply tabs/utils/file-organizer.sh:76]: I am not debating friend, I just want to learn why we should be calling those functions if we are not going to use them anywhere. For example calling checkAUR is not needed here and checksuper group is also not useful here either.

Comment (guruswarupa) [reply tabs/utils/file-organizer.sh:76]: > I am not debating friend, I just want to learn why we should be calling those functions if we are not going to use them anywhere. For example calling checkAUR is not needed here and checksuper group is also not useful here either.

i had same doubt but i had to use it.

Comment (jeevithakannan2) [reply tabs/utils/file-organizer.sh:76]: > > I am not debating friend, I just want to learn why we should be calling those functions if we are not going to use them anywhere. For example calling checkAUR is not needed here and checksuper group is also not useful here either.
> 
> i had same doubt but i had to use it.

See I have the same doubt that's why I was asking him

**Comment by ChrisTitusTech** [comment]
A bit too obscure and there are better GUI tools for this.

**Pull request closed without merging**